discovery/api/v1alpha1/addon_types.go

```diff
@@ -18,18 +18,30 @@ package v1alpha1
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 // AddonSpec defines the desired state of Addon
 type AddonSpec struct{}
 
 // AddonStatus describes the observed state of an addon and its components.
 type AddonStatus struct {
+	// Metadata extracts information about the targeted objects based on labels.
+	// +optional
+	Metadata []Metadata `json:"metadata,omitempty"`
+
 	// Components describes resources that compose the addon.
 	// +optional
 	Components *Components `json:"components,omitempty"`
 }
 
+// Metadata describes the content extracted from a target object.
+type Metadata struct {
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
Does this make the schema non-structural? or is that only when `PreserveUnknownFields` is schema-wide?